opm/autodiff/RateConverter.hpp

```diff
@@ -450,7 +450,66 @@ namespace Opm {
                 }   
                 calcRmax();
             }
+
             /**
+             * Compute average hydrocarbon pressure and maximum
+             * dissolution and evaporation at average hydrocarbon
+             * pressure in all regions in field.
+             *
+             * Fluid properties are evaluated at average hydrocarbon
+             * pressure for purpose of conversion from surface rate to
+             * reservoir voidage rate.
+             *
+             * \param[in] state Dynamic reservoir state.
+             * \param[in] any  The information and communication utilities
+             *                 about/of the parallelization. in any parallel
+             *                 it wraps a ParallelISTLInformation. Parameter
+             *                 is optional.
+             */
+            template <typename ElementContext, class EbosSimulator>
+            void defineState(const EbosSimulator& simulator)
+            {
+
+                //const int numCells = cellPvtIdx_.size();
+                //const Region region = std::vector<int>(numCells, 0);
+                auto& ra = attr_.attributes(0);
```

Comment:
I have been looking at this code for a while. I still cannot see how the new function `RateConverter::defineState(EbosSimulator)` calculates averages over the regions. For me it seems like the average is done once over the whole domain. What am I missing?

Reply:
I might remember wrong. I am not sure we support averages over separate regions.

Reply:
I got the impression form the following code from StandardWellsDense_impl.hpp. (line number is different from master.)
```
787                 // the average hydrocarbon conditions of the whole field will be used
788                 const int fipreg = 0; // Not considering FIP for the moment.
789 
790                 rate_converter_.calcCoeff(well_rates, fipreg, convert_coeff);
791                 well_voidage_rates[w] = std::inner_product(well_rates.begin(), well_rates.end(),
792                                                            convert_coeff.begin(), 0.0)
```

Reply:
You are right. It doesn't. We only support calculating averages over the whole domain. If option 13 is set in WELSPECS. The averages are supposed to be calculated for each FIP region. I am currently debugging the RateConverter.hpp. There are several things in this code that may not be correct.